src/Sylius/Component/Taxation/Resolver/TaxRateResolver.php

```diff
@@ -13,14 +13,18 @@
 
 namespace Sylius\Component\Taxation\Resolver;
 
+use Sylius\Calendar\Provider\DateTimeProviderInterface;
 use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Sylius\Component\Taxation\Checker\TaxRateDateEligibilityCheckerInterface;
 use Sylius\Component\Taxation\Model\TaxableInterface;
 use Sylius\Component\Taxation\Model\TaxRateInterface;
 
 class TaxRateResolver implements TaxRateResolverInterface
 {
-    public function __construct(protected RepositoryInterface $taxRateRepository)
-    {
+    public function __construct(
+        protected RepositoryInterface $taxRateRepository,
+        protected ?TaxRateDateEligibilityCheckerInterface $taxRateDateChecker = null
```

Comment:
Add a note to UPGRADE file ([example](https://github.com/Sylius/Sylius/blob/1.12/UPGRADE-1.10.md#upgrade-from-v1108-to-v11010)) about this change and trigger deprecation if there is no second argument passed ([example](https://github.com/Sylius/Sylius/blob/1.12/src/Sylius/Component/Core/Cart/Context/ShopBasedCartContext.php#L48:L50)).